src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/LicenseScanTests.cs

```diff
@@ -0,0 +1,243 @@
+        "gpl-2.0 WITH gcc-linking-exception-2.0", // https://github.com/nexB/scancode-toolkit/blob/develop/src/licensedcode/data/rules/gpl-2.0_with_gcc-linking-exception-2.0_6.RULE
+        "isc", // https://opensource.org/license/isc-license-txt/
+        "json", // https://www.json.org/license.html
+        "lgpl-2.0-plus", // https://opensource.org/license/lgpl-2-0/
+        "lgpl-2.1", // https://opensource.org/license/lgpl-2-1/
+        "lgpl-2.1-plus", // https://opensource.org/license/lgpl-2-1/
+        "mit", // https://opensource.org/license/mit/
+        "mit-addition", // https://github.com/nexB/scancode-toolkit/blob/develop/src/licensedcode/data/licenses/mit-addition.LICENSE
+        "ms-pl", // https://opensource.org/license/ms-pl-html/
+        "ms-rl", // https://opensource.org/license/ms-rl-html/
+        "newton-king-cla", // https://github.com/nexB/scancode-toolkit/blob/develop/src/licensedcode/data/licenses/newton-king-cla.LICENSE
+        "ngpl", // https://opensource.org/license/nethack-php/
+        "ofl-1.1", // https://opensource.org/license/ofl-1-1/
+        "osf-1990", // https://fedoraproject.org/wiki/Licensing:MIT?rd=Licensing/MIT#HP_Variant
+        "public-domain", // https://github.com/nexB/scancode-toolkit/blob/develop/src/licensedcode/data/licenses/public-domain.LICENSE
+        "public-domain-disclaimer", // https://github.com/nexB/scancode-toolkit/blob/develop/src/licensedcode/data/licenses/public-domain-disclaimer.LICENSE
+        "rpl-1.5", // https://opensource.org/license/rpl-1-5/
+        "unicode", // https://opensource.org/license/unicode-inc-license-agreement-data-files-and-software/
+        "uoi-ncsa", // https://opensource.org/license/uoi-ncsa-php/
+        "warranty-disclaimer", // https://github.com/nexB/scancode-toolkit/blob/develop/src/licensedcode/data/licenses/warranty-disclaimer.LICENSE
+        "zlib" // https://opensource.org/license/zlib/
+    };
+
+    private static readonly string[] s_ignoredFilePatterns = new string[]
+    {
+        "*.bin",
+        "*.bmp",
+        "*.bson",
+        "*.db",
+        "*.dic",
+        "*.docx",
+        "*.eot",
+        "*.gif",
+        "*.ico",
+        "*.jpg",
+        "*.il",
+        "*.ildump",
+        "*.lss",
+        "*.nlp",
+        "*.otf",
+        "*.pdf",
+        "*.pfx",
+        "*.png",
+        "*.rtf",
+        "*.snk",
+        "*.ttf",
+        "*.vsd",
+        "*.vsdx",
+        "*.winmd",
+        "*.woff",
+        "*.woff2",
+        "*.xlsx",
+    };
+
+    private readonly string _targetRepo;
+
+    public LicenseScanTests(ITestOutputHelper outputHelper) : base(outputHelper)
+    {
+        Assert.NotNull(Config.LicenseScanPath);
+        _targetRepo = new DirectoryInfo(Config.LicenseScanPath).Name;
+    }
+
+    [SkippableFact(Config.LicenseScanPathEnv, skipOnNullOrWhiteSpaceEnv: true)]
```

Comment:
If I understand this test correctly, there are multiple settings to control whether or not this test runs (e.g. LicenseScanPathEnv and category==LicenseScan).  In general I don't think this provides a great UX for folks trying to get the test to run.  Missing one ends up in nothing running.  This could cause us to be in a situation where we think something is configured to run in CI but it isn't.  IMO there should be one setting to control the running.  If the required ENVs are not set, the test should fail with a clear message that indicates which ENVs are required but not specified.

Reply:
Fixed in https://github.com/dotnet/installer/pull/17442/commits/b07e92576de369a531ddeb9b212348b21dc1b050